Topic1/key5/Offer.java

```diff
@@ -0,0 +1,66 @@
+package Topic1.key5;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import Topic1.key1.Cart;
+import Topic1.key1.Item;
+import Topic1.key1.Product;
+
+/**
+ * @author Jordan
+ * an offer includes other individual products or other offers, 
+ * his price and name must be setted independently,
+ *  if you want to know the real price, you�ll need to use the calculateRealPrice() procedure 
+ */
+public class Offer extends Product{
+	protected ArrayList<Product> productsOffer;
+	//inhirids name and price
+	
+	public Offer(ArrayList<Product> productsOffer, String name, String price) {
+		super();
+		this.productsOffer = productsOffer;
+		this.price=new BigDecimal(price); //offer price is independent of the calculatedPrice
```

Comment:
This conversion aspect is being propagated all over the code. We'll need to keep it in one place, or even we might remove it in the future.